x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { get } from 'lodash';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiFieldNumber,
+  EuiText,
+  EuiRange,
+  EuiSwitch,
+} from '@elastic/eui';
+import { EuiIconTip } from '@elastic/eui';
+import {
+  isAutoInterval,
+  IFieldFormat,
+  UI_SETTINGS,
+} from '../../../../../../../../src/plugins/data/common';
+import { MODES, RangeColumnParams, UpdateParamsFnType, MODES_TYPES } from './ranges';
+import { AdvancedRangeEditor } from './advanced_editor';
+
+const BaseRangeEditor = ({
+  autoIntervalEnabled,
+  maxBars,
+  interval,
+  maxHistogramBars,
+  onToggleEditor,
+  toggleAutoInterval,
+  onMaxBarsChange,
+  onIntervalChange,
+}: {
+  autoIntervalEnabled: boolean;
+  maxBars: 'auto' | number;
+  interval: number;
+  maxHistogramBars: number;
+  onToggleEditor: () => void;
+  toggleAutoInterval: (enabled: boolean) => void;
+  onMaxBarsChange: (newMaxBars: number) => void;
+  onIntervalChange: (newInterval: number) => void;
+}) => {
+  // TODO: manage the temporary empty string for interval
+  const sectionLabel = i18n.translate('xpack.lens.indexPattern.ranges.granularity', {
+    defaultMessage: 'Granularity',
+  });
+
+  return (
+    <>
+      <EuiFormRow
+        label={sectionLabel}
+        data-test-subj="indexPattern-ranges-section-label"
+        labelAppend={
+          <EuiSwitch
+            label={i18n.translate('xpack.lens.indexPattern.ranges.autoInterval', {
+              defaultMessage: 'Auto interval',
+            })}
+            checked={autoIntervalEnabled}
+            onChange={(e) => toggleAutoInterval(e.target.checked)}
+            data-test-subj="indexPattern-ranges-auto-interval"
+            compressed
+          />
+        }
+      >
+        <>
+          <EuiFlexGroup gutterSize="s" responsive={false} wrap>
+            <EuiFlexItem>
```

Comment:
This flex and fragment cause issues with a11y because input need to be direct descendents of EuiFormRow. Only a single input is shown in this configuration at a time so there's no need for flex display.


```suggestion
```